docs/cni-plugin.md

```diff
@@ -92,3 +92,87 @@ This also needs `host-local` ipam plugin to be present in one of the `PATH` dire
 ```shell
 sudo --preserve-env make test-pkg-plugin
 ```
+
+## OVS Hardware Offload
+
+This provides an option to enable OVS hardware offload with ovs-cni while using OVS data-plane
+in Mellanox ConnectX-4 Lx onwards NIC hardware (Mellanox Embedded Switch or eSwitch).
+The OVS hardware offload is using SR-IOV technology with VF representor host net-device.
+The VF representor plays the same role as TAP devices in Para-Virtual (PV) setup.
+A packet sent through the VF representor on the host arrives to the VF, and a packet sent
+through the VF is received by its representor.
+
+### Prerequisites
+
+[sriov-network-device-plugin](https://github.com/intel/sriov-network-device-plugin)
+[multus-cni](https://github.com/intel/multus-cni)
+[network-resources-injector](https://github.com/intel/network-resources-injector)
+
+### Mellanox SR-IOV Configuation
+
+To enable OVS hardware offloading, create VFs from a Mellanox PF, configure VFs in `switchdev` mode and
+set `hw-offload=true` for Open vSwitch.
+For more information [refer Mellanox doc] (https://www.mellanox.com/related-docs/prod_software/ASAP2_Hardware_Offloading_for_vSwitches_User_Manual_v4.4.pdf)
+
+### Device Plugin configuation
+
+The device plugin would create the resouce pools based on the configurations given in the `/etc/pcidp/config.json`.
```

Comment:
who is expected to create this file properly? would  https://github.com/openshift/sriov-network-operator create something acceptable?

Reply:
There is a close connection between the configuration of (SRIOV) VF device pools and the creation of NADs referring to these as resources (for automatic injection of resources into the pod spec through the network resources injector (https://github.com/intel/network-resources-injector). For a given Mellanox SmartNICs with LAG, all VFs will be typically form one single pool as the they are fully "programmable" as OVS interfaces at pod creation. 

So a typical setup would be CX-5 SmartNIC with two bonded PFs  associated with one OVS bridge and one VF pool. On top of that there could be any number of ovs-cni NADs for individual VLAN or VLAN trunk ports on that bridge.

We have been looking at the openshift sriov-network-operator before, but found it not fulfilling our needs fully. Also there were some dependencies on the openshift environment that made it hard to get it to work on vanilla K8s.

I think we should look into the provisioning of networks and associated network device pools coherently in one effort.